src/main/java/com/example/service/impl/BookServiceImpl.java

```diff
@@ -32,8 +31,25 @@ public List<BookDto> getAll() {
 
     @Override
     public BookDto getById(Long id) {
-        Book book = bookRepository.getById(id).orElseThrow(
-                () -> new EntityNotFoundException("DB don't have book with such id = " + id));
+        Book book = bookRepository.getById(id);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        Book book = bookRepository.getReferenceById(id);
```

Comment:
https://www.baeldung.com/spring-data-jpa-getreferencebyid-findbyid-methods
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(()...
```